--- a/net/minecraft/client/resources/model/ModelDiscovery.java
+++ b/net/minecraft/client/resources/model/ModelDiscovery.java
@@ -60,6 +_,7 @@
         this.referencedModels.put(SpecialModels.BUILTIN_GENERATED, SpecialModels.GENERATED_MARKER);
         this.referencedModels.put(SpecialModels.BUILTIN_BLOCK_ENTITY, SpecialModels.BLOCK_ENTITY_MARKER);
         Set<ModelResourceLocation> set = listMandatoryModels();
+        net.minecraftforge.client.ForgeHooksClient.onRegisterAdditionalModels(set);
         p_361458_.models().forEach((p_366538_, p_363254_) -> {
             this.registerTopModel(p_366538_, p_363254_.model());
             set.remove(p_366538_);
